### ISSUE TITLE
Example code not working

### ISSUE THREAD

**Issue by lazamar** [kazu-yamamoto/http2#29]
Hi, thanks for the work on this library. I am trying to get started with it but the example code is failing.

When I run it in my OSX Big Sur 11.4 with Stack 2.5.1 I get

```
> stack run client google.com 80
client: thread blocked indefinitely in an MVar operation
```

When running on a redhat machine with Stack version 2.7.1  and Red Hat 8.2.1 I get

```
 > https_proxy=fwdproxy:8080 http_proxy=fwdproxy:8080 stack run client lazamar.co.uk 80
client: Network.Socket.connect: <socket: 11>: does not exist (Network is unreachable)
```

Is there anything else required for the example to work?

**Comment by kazu-yamamoto** [comment]
`util/client.hs` does not support proxy.

**Comment by lazamar** [comment]
The second example was run on my work computer. The first one, however, (`stack run client google.com 80`) was on my personal computer.
When you run the example code in https://hackage.haskell.org/package/http2-3.0.2/docs/Network-HTTP2-Client.html does it succeed?

**Comment by kazu-yamamoto** [comment]
- The example in the client document has been fixed via 7036a3429fb08bfcd5947230c37d1f3e63dfb3a6.
- `client` was renamed to `h2c-client` which runs HTTP2 over TCP.
- google.com 80 does not support HTTP/2 over TCP. It's HTTP/1.1 over TCP only.
- `h2-client` in `http2-tls`, which runs HTTP2 over TLS, can talk with google.com:443.
